code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -1300,14 +1301,15 @@ obj/machinery/computer/pandemic/proc/replicator_cooldown(var/waittime)
 	var/state_change = 0
 	if(on)
 		if(beaker)
-			if(beaker.reagents.chem_temp > temperature)
-				beaker.reagents.chem_temp = max(beaker.reagents.chem_temp-rate, temperature)
-				beaker.reagents.handle_reactions()
-				state_change = 1
-			else if(beaker.reagents.chem_temp < temperature)
-				beaker.reagents.chem_temp = min(beaker.reagents.chem_temp+rate, temperature)
+			if(beaker.reagents.chem_temp != desired_temp)
+				beaker.reagents.chem_temp += ((desired_temp - beaker.reagents.chem_temp) * heater_coefficient)
+
+				beaker.reagents.chem_temp = round(beaker.reagents.chem_temp) //stops stuff like 456.12312312302
+				if(beaker.reagents.chem_temp == ((desired_temp-3) || (desired_temp+3)))//need better way to stop superdecimals and rounding drops

```

Comment:
Err, sorry, I should have phrased it better. I've been using 'error' in the (outdated, but common) scientific meaning of 'uncertainty', not 'failure'. I've been spending too much time writing lab reports lately.

No machine, anywhere, is perfectly accurate - and as far as heating components go, two ovens made at the same time and used for a while can have wildly differing actual temperatures even if you set them to the same temperature. So if you set a RL heater to go to 200 degrees, it may actually be at 195 or 205 degrees at any given moment if it has a +/- 5 degrees tolerance.

And as soon as you have the UI that lets you check the temperature in real-ish time, this would mean that if you made a reaction that turns a reagent into ashes or makes it explode within that range, a chemist would have to actually pay attention to what he's doing and adjust the temperature if it goes too high. So someone with MAD CH3M1STRY SKILLZ could make any chem in his heater made from soda cans, floor tiles and prayer by knowing how and when to adjust it, but a ChemHeater9000K means Assistant McHonf can just drop the beaker in, set temperature and go get a coffee.